packages/shared/src/components/modals/streaks/StreakRecoverModal.tsx

```diff
@@ -0,0 +1,226 @@
+        fill="#FC538D"
+      />
+    </svg>
+  </div>
+);
+
+const StreakRecoverHeading = ({ days }: { days: number }) => (
+  <Typography
+    className="-mx-1 -mb-1 text-center font-bold"
+    tag={TypographyTag.H3}
+    type={TypographyType.Title1}
+    data-testid="streak-recover-modal-heading"
+  >
+    Oh no! Your
+    <span className="text-accent-bacon-default"> {days} day streak </span> has
+    been broken!
+  </Typography>
+);
+
+const StreakRecoveryCopy = ({
+  reputation,
+  cost,
+}: Record<'reputation' | 'cost', number>) => {
+  const isFree = cost === 0;
+  const canRecover = reputation >= cost;
+  const reputationLink = (
+    <a
+      target="_blank"
+      rel={anchorDefaultRel}
+      href={reputationHref}
+      title="What are reputation points?"
+      className="text-text-link underline-offset-2 hover:underline"
+    >
+      reputation points
+    </a>
+  );
+
+  return (
+    <Typography
+      className="text-center"
+      tag={TypographyTag.P}
+      type={TypographyType.Body}
+      data-testid="streak-recovery-copy"
+    >
+      {isFree && (
+        <>
+          Lucky you! The first streak restore is on us 🎁. This usually costs 25{' '}
+          {reputationLink}. Be sure to come prepared next time!
+        </>
+      )}
+      {!isFree &&
+        (canRecover ? (
+          <>
+            Maintain your streak!
+            <br />
+            Use {cost} {reputationLink} to keep going.
+          </>
+        ) : (
+          <>You don’t have enough {reputationLink} to restore your streaks.</>
+        ))}
+    </Typography>
+  );
+};
+
+const StreakRecoverButton = ({
+  cost,
+  ...props
+}: Record<'cost', number> & ButtonProps<'button'>) => (
+  <Button
+    {...props}
+    className="relative"
+    variant={ButtonVariant.Primary}
+    size={ButtonSize.Large}
+    data-testid="streak-recover-button"
+  >
+    Restore my streak
+    <span className="absolute -right-0.5 -top-0.5 flex items-center whitespace-nowrap rounded-bl-14 rounded-tr-14 bg-accent-onion-default px-1.5 py-1 text-text-primary">
+      <Typography
+        tag={TypographyTag.Span}
+        type={TypographyType.Caption1}
+        data-testid="streak-recovery-cost"
+      >
+        {cost} Rep
+      </Typography>{' '}
+      <ReputationIcon className="inline-block" size={IconSize.Size16} />
+    </span>
+  </Button>
+);
+
+export const StreakRecoverModal = (
+  props: StreakRecoverModalProps,
+): ReactElement => {
+  const { isOpen, onRequestClose, user } = props;
+
+  const id = useId();
+  const { recover, hideForever, onClose, onRecover } = useStreakRecover({
+    onRequestClose,
+  });
+
+  if (!user || !recover.canDo || recover.isLoading || recover.isDisabled) {
```

Comment:
how does the `recover.canDo` and `canUserRecover` calculated below change?

Reply:
`recover.canDo` is coming from be, `canUserRecover` is much more `canUserAffordTheCost`, maybe we want to change the name of that variable 👀

Reply:
recover.canDo -> comes from be yes but what does it mean in this case?

Reply:
We use it to check if the user still can do the recover at the moment of the popup showing, if is false the component just return null. 
With @sshanzel today we saw that this may be a good point for block popup from showing and update the `Alerts` table for not making too much controls on cron.

Reply:
But recover.canDo also checks if user can afford the cost right? we're checking here again to help any edge case where user has less reputation then when we fetched the data recover.canDo?

Reply:
No currently do not check if the user can afford the cost, only check if is eligible, just because we want to show the popup with "you can recover the streak but currently not have enough point" if the user cannot afford it. 
Maybe `recover.isEligible` is a better name?

Reply:
hmm maybe, both give the idea that that check might already be complete that's why the confusion on my side. What makes the user eligible in this case? The fact he only broke the streak today (1day - 24h?)? anything else?

Reply:
User can see the popup if lost streak in the past day, but can recover only if have enough points. 
We can differentiate with a `recover.isVisible`? But is that name ok from the be?

Reply:
I wouldn't worry at this point anymore, think what we have is ok at the moment. 👍